content/blog/gain-platform-superpowers-with-the-kebap-stack/index.md

```diff
@@ -0,0 +1,704 @@
+orchestrated and managed through Pulumi's infrastructure as code capabilities.
+
+Platform engineering addresses these challenges by creating a layer of abstraction between development teams and the
+underlying infrastructure. This approach aims to simplify workflows, standardize practices, and empower developers to
+focus on creating business value rather than grappling with intricate infrastructure details.
+
+The [Cloud Native Computing Foundation](https://www.cncf.io/) (CNCF) landscape offers a rich ecosystem of tools and
+technologies that can be
+leveraged to build these platforms. However, the sheer number of options can be overwhelming, making it difficult for
+organizations to determine the best combination of tools for their specific needs.
+
+At Pulumi's Customer Experience Team, we've observed that adopting infrastructure as code often serves as a catalyst for
+organizations to reassess their entire software delivery process. This reassessment frequently leads to the
+implementation of more streamlined, automated, and secure methodologies – the core principles of platform engineering.
+
+To address these challenges and provide guidance, we've created a reference archticture called the `KEBAP` stack.
+Demonstrating how various CNCF technologies can
+be integrated cohesively using Pulumi to create a robust, scalable internal developer platform.
+
+By leveraging `KEBAP` and the wider CNCF ecosystem, organizations can accelerate their journey towards effective
+platform
+engineering. This approach not only enhances operational efficiency but also positions teams to readily adopt and
+benefit from emerging Cloud Native technologies, ensuring their platforms remain future-proof and aligned with industry
+best practices.
+
+## One Reference Architecture: The `KEBAP` Stack
+
+The `KEBAP` stack consists of the following components:
+
+- **K**ubernetes (with Kyverno as a bonus)
+- **E**xternal Secrets Operator
+- **B**ackstage
+- **A**rgo CD
+- **P**ulumi
+
+This combination forms a standardized GitOps stack applicable to various infrastructure projects. By implementing the
+`KEBAP` stack, we've consistently reduced the time required for teams to establish development and production
+environments from months to minutes.
+
+To following along, we are going to use
+the [capabilities of platforms](https://tag-app-delivery.cncf.io/whitepapers/platforms/#capabilities-of-platforms)
+definition from the CNCF working group on Platforms.
+
+<img src="base.png">
+
+## Let's Break It Down
+
+Let's take a closer look at each component of the `KEBAP` stack and understand how they work together to streamline the
+software delivery process.
+
+The five layers of the `KEBAP` stack are designed to address different aspects of the software delivery process:
+
+### Kubernetes
+
+<img src="k.png">
+
+[Kubernetes](https://kubernetes.io/) serves as the central control plane. It provides a well-defined API for managing
```

Comment:
This is the part that doesn't make sense to me. Is K8s actually the control plane. K8s is the place that manages the compute workloads that is for sure. In some ways really K8s is the data plane part of a platform right?

Reply:
k8s is referred in this context as control plane. Especially when working with controllers that provision stuff like Databases, which run on k8s too.